### PR TITLE
Delegate "info" calls to wrapped Reporter

### LINKED ISSUE #8: Clippy 0.2.2 breaks Quill
Related to #6, when using both the Clippy plugin and Quill, the compilation is broken. Easily reproducible with any of [those examples](http://getquill.io).

```sbt
[error] ~/project/src/main/scala/Main.scala:17: exception during macro expansion: java.lang.StackOverflowError
[error]   db.run(query)(CSVSources.data.flatMap(_.toOption))
```

### PR DESCRIPTION
**Pull request by kciesielski**
- This fixes #8

### CODE REVIEW

**Comment by kciesielski** [comment]
Not sure why the build fails. It says
`The command "[ "${TRAVIS_PULL_REQUEST}" = "false" ] && sbt updateImpactSubmit || false" exited with 1.`

File: plugin/src/main/scala-2.11/com/softwaremill/clippy/DelegatingReporter.scala [hunk 1]
```diff
@@ -7,7 +7,7 @@ class DelegatingReporter(r: Reporter, handleError: (Position, String) => String)
   override protected def info0(pos: Position, msg: String, severity: Severity, force: Boolean) = {
     // cannot delegate to info0 as it's protected, hence special-casing on the possible severity values
     if (severity == INFO) {
-      info(pos, msg, force)
+      r.info(pos, msg, force)
```
Comment (adamw) [inline plugin/src/main/scala-2.11/com/softwaremill/clippy/DelegatingReporter.scala]: shouldn't we do the same for warning, error?

Comment (kciesielski) [inline plugin/src/main/scala-2.11/com/softwaremill/clippy/DelegatingReporter.scala]: We do it for warning:
````scala 
override def warning(pos: Position, msg: String) = 
  r.warning(pos, msg)
```
and for error
```scala  
override def error(pos: Position, msg: String) = 
  r.error(pos, handleError(pos, msg))
```

similarly for `echo()`

Comment (adamw) [inline plugin/src/main/scala-2.11/com/softwaremill/clippy/DelegatingReporter.scala]: true, it just didn't look this way with the code folded :)

**Pull request merged**